vcd/resource_vcd_edgegateway.go

```diff
@@ -206,23 +210,38 @@ func resourceVcdEdgeGatewayRead(d *schema.ResourceData, meta interface{}) error
 
 	vcdClient := meta.(*VCDClient)
 
-	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "name")
+	identifier := d.Id()
+	if identifier == "" {
```

Comment:
why identifier can be empty?

Reply:
This read method is used for both the resource and the data source. When called from the data source, `d.Id()` is empty.